ethsigner/signer/multifile-based/src/main/java/tech/pegasys/ethsigner/signer/multifilebased/KeyPasswordLoader.java

```diff
@@ -42,31 +42,43 @@
   }
 
   Optional<KeyPasswordFile> loadKeyAndPasswordForAddress(final String address) {
-    final String filename = normalizeAddress(address);
-    final Path keyPath = keysDirectory.resolve(filename + KEY_FILE_EXTENSION);
-    if (keyPath.toFile().exists() && isFilenameValid(keyPath)) {
-      return tryFindingMatchingPassword(keyPath).map(path -> new KeyPasswordFile(keyPath, path));
-    } else {
+    final List<KeyPasswordFile> matchingKeys =
+        loadAvailableKeys().stream()
+            .filter(kp -> kp.getName().toLowerCase().contains(normalizeAddress(address)))
```

Comment:
Do we want this to be contains? The tests are only for a prefix,  so would expect this be be a endsWith excluding the file extension

Reply:
Thanks! I've also added tests for this now.
`loadKeyPasswordNotEndingWithAddressReturnsEmpty`
`loadAvailableKeysWithPasswordNotEndingWithAddressReturnsEmptyList`